src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -70,9 +74,28 @@ get_cu_status(uint32_t cu_status)
   return pt;
 }
 
+static void
+schedulerUpdateStat(xrt_core::device *device)
+{
+  try {
+    // lock xclbin
+    auto uuid = xrt::uuid(xrt_core::device_query<xrt_core::query::xclbin_uuid>(device));
+    device->open_context(uuid.get(), static_cast<unsigned int>(-1), true);
+    auto at_exit = [] (auto device, auto uuid) { device->close_context(uuid.get(), static_cast<unsigned int>(-1)); };
+    xrt_core::scope_guard<std::function<void()>> g(std::bind(at_exit, device, uuid));
+    
+    device->update_scheduler_status();
+  }
+  catch (const std::exception&) {
+    // xclbin_lock failed, safe to ignore
+  }
+}
+
 boost::property_tree::ptree
 populate_cus(const xrt_core::device *device)
 {
+  if (!std::getenv("XCL_SKIP_CU_READ"))

```

Comment:
Could you help me understand why we are looking at environment variables.  Unfortunately, this is a bad practice and should be avoided whenever possible.

Reply:
Not needed. removed